src/modeler/AutoRest.Swagger/Validation/APIVersionPattern.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using System.Text.RegularExpressions;
+
+namespace AutoRest.Swagger.Validation
+{
+    /// <summary>
+    /// Validates the version of the swagger. API version must follow the date pattern
+    /// yyyy-MM-dd and allowed prefixes are -preview, -alpha, -beta, -rc, -privatepreview.
+    /// </summary>
+    public class APIVersionPattern : TypedRule<string>
+    {
+        private static readonly Regex VersionRegex = new Regex(@"^(20\d{2})-(0[1-9]|1[0-2])-((0[1-9])|[12][0-9]|3[01])(-(preview|alpha|beta|rc|privatepreview))?$");
```

Comment:
Hey @sarangan12 @ravbhatnagar - Please check this out ["version": "2017-01-01.4.0",](https://github.com/Azure/azure-rest-api-specs/pull/892/files#diff-f89b95611b96682df64fd1ca46698865R5)

The api-version format is a little different. I think the regex will not accommodate that. 
Would like to know what is the guidance from ARM team regarding this.